internal/status/gateway.go

```diff
@@ -50,6 +50,10 @@ func UpdateGatewayStatusProgrammedCondition(gw *gwapiv1b1.Gateway, svc *corev1.S
 			}
 		}
 
+		if svc.Spec.Type == corev1.ServiceTypeNodePort {
+			addresses = nodeAddresses
```

Comment:
should we also append the NodePort to this address, and expose it in the status ?

Reply:
looks like we cannot expose the port in a clean way, because there will be a unique node port per listener port